katzenmint/state.go

```diff
@@ -383,15 +384,31 @@ func (state *KatzenmintState) GetDocument(epoch uint64, height int64) ([]byte, *
  *               Document                *
  *****************************************/
 
+func (s *KatzenmintState) pruneDocument() {
+	begin := storageKey(descriptorsBucket, []byte{}, s.currentEpoch)
+	end := storageKey(descriptorsBucket, []byte{}, s.currentEpoch+1)
+	_ = s.tree.IterateRange(begin, end, true, func(key, value []byte) (ret bool) {
+		_, err := s11n.ParseDescriptorWithoutVerify(value)
+		if err != nil {
+			s.tree.Remove(key)
+		}
+		return false
+	})
+}
+
 func (s *KatzenmintState) generateDocument() (*document, error) {
 	// Cannot lock here
 
 	// Load descriptors (providers and nodesDesc).
 	var providersDesc, nodesDesc []*descriptor
 	begin := storageKey(descriptorsBucket, []byte{}, s.currentEpoch)
 	end := storageKey(descriptorsBucket, []byte{}, s.currentEpoch+1)
-	_ = s.tree.IterateRange(begin, end, true, func(key, value []byte) (ret bool) {
-		desc, _ := s11n.ParseDescriptorWithoutVerify(value)
+	escaped := s.tree.IterateRange(begin, end, true, func(key, value []byte) (ret bool) {
+		desc, err := s11n.ParseDescriptorWithoutVerify(value)
+		if err != nil {
+			// TODO: check different error
```

Comment:
Can you be more precise here with what needs to be the actual error and if it's something more involved and complex, you should open an issue

Reply:
The error might be corrupted data or invalid certificate (outdated/incorrect encoding). Because the network was stuck or offline (still need to find the root cause), the descriptors might be expired in local database (saved before generate document). How about we remove descriptors and wait them submit again?

Reply:
Let's do that

Reply:
@Mikerah is this possible to generate different block hash for old blocks with expired certificate?